custom/abdm/commcare_extensions.py

```diff
@@ -0,0 +1,18 @@
+from casexml.apps.phone.xml import get_custom_user_data_for_restore
+
+from corehq.toggles import ABDM_INTEGRATION
+from custom.abdm.models import ABDMUser
+
+
+def _get_abdm_api_token(username, domain):
+    user, _ = ABDMUser.objects.get_or_create(username=username, domain=domain)
+    user.generate_token()
+    user.save()
```

Comment:
This pattern of modifying data in fetch/get calls should be used with caution. Consider refactoring the code to avoid this pattern.